renku/service/views/datasets.py

```diff
@@ -0,0 +1,325 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service datasets view."""
+import json
+import os
+
+from flask import jsonify, request
+
+from renku.core.commands.dataset import add_file, create_dataset, \
+    dataset_parent, list_files
+from renku.service.config import INTERNAL_FAILURE_ERROR_CODE, \
+    INVALID_PARAMS_ERROR_CODE
+from renku.service.views import find_file, repo_sync, requires_identity, \
+    validate_project_id
+
+
+@requires_identity
+def list_datasets_view(user):
+    """
+    List all datasets in specified project id.
+
+    ---
+    get:
+        tags:
+        - "datasets"
+
+        summary: List all datasets within a project.
+        description: List all datasets within a project.
+
+        parameters:
+            - name: project_id
+              in: query
+              description: Project identifier.
+              schema:
+                type: string
+              required: true
+
+        responses:
+            result:
+                description: |
+                  Contains `datasets` key with
+                  list of all datasets from a specified project.
+
+            error:
+                description: |
+                  Contains `code` and `message`
+                  describing the error.
+
+    """
+    if request.method == 'GET':
+        project_id = request.args.get('project_id')
+        pwd = validate_project_id(project_id)
+        if not pwd:
+            return jsonify(status=2, message='invalid project_id argument')
+
+        cwd = os.getcwd()
+        os.chdir(str(pwd))
```

Comment:
Because `LocalClient` is not publicly exposed to the interface. That abstraction is heavily broken and should not be used outside of the internals of the core. The core module exposes the interface to work with which are designed for this purpose. We could think of adding cd as an optional flag to those methods.